packages/rust/server/tests/helper.rs

```diff
@@ -0,0 +1,246 @@
+
+use axum::http::StatusCode;
+use bubble::models::confirmation::Confirmation;
+
+use bubble::routes::user::{ChangeEmail, Confirm, CreateUser, SessionToken, SignInJson};
+
+use bubble::models::session::Session;
+use sqlx::migrate::MigrateDatabase;
+use sqlx::Postgres;
+use uuid::Uuid;
+
+pub struct TempDatabase {
+    pool: DbPool,
+    _db_url: String,
+}
+
+impl TempDatabase {
+    pub async fn new() -> Self {
+        let db_url = format!("{}/{}", env::var("DB_URL").unwrap(), Uuid::new_v4());
+        Postgres::create_database(&db_url).await.unwrap();
+        let pool = PgPoolOptions::new()
+            .max_connections(5)
+            .connect(&db_url)
+            .await
+            .unwrap();
+
+        sqlx::migrate!("../db/migrations").run(&pool).await.unwrap();
+
+        Self {
+            pool,
+            _db_url: db_url,
+        }
+    }
+
+    pub fn pool(&self) -> &DbPool {
+        &self.pool
+    }
+}
+
+pub async fn start_server(pool: DbPool) -> TestClient {
+    let router = router::router(pool);
+
+    TestClient::new(router)
+}
+
+// For user authentication testing only
+pub async fn signup_user(
+    db: &DbPool,
+    client: &TestClient,
+    user_in: &CreateUser,
+) -> Result<(User, Uuid), StatusCode> {
+    let res = client
+        .post("/user/signup")
+        .header("Content-Type", "application/json")
+        .body(serde_json::to_string(user_in).unwrap())
+        .send()
+        .await;
+    assert_eq!(res.status(), StatusCode::CREATED);
+
+    let user = User::from_username(db, &user_in.username).await.unwrap();
+    let link_id = Confirmation::filter_user_id(db, user.id).await.unwrap()[0].link_id;
+
+    Ok((user, link_id))
+}
+
+// For user authentication testing only
+pub async fn signup_confirm_user(
+    db: &DbPool,
+    client: &TestClient,
+    confirm: &Confirm,
+    user_in: &User,
+) -> Result<(User, Uuid), StatusCode> {
+    let res = client
+        .post("/user/signup-confirm")
+        .header("Content-Type", "application/json")
+        .body(serde_json::to_string(confirm).unwrap())
+        .send()
+        .await;
+    assert_eq!(res.status(), StatusCode::CREATED);
+
+    let user = User::from_id(db, user_in.id).await.unwrap();
+    let token: SessionToken = res.json().await;
+    Ok((user, Uuid::parse_str(&token.token).unwrap()))
+}
+
+pub async fn signin_user(
+    _db: &DbPool,
+    client: &TestClient,
+    user: &User,
+) -> Result<Uuid, StatusCode> {
+    let signin = SignInJson {
+        email: user.email.clone().unwrap(),
+        password: user.password.clone(),
+    };
+    let res = client
+        .post("/user/signin")
+        .header("Content-Type", "application/json")
+        .body(serde_json::to_string(&signin).unwrap())
+        .send()
+        .await;
+    assert_eq!(res.status(), StatusCode::CREATED);
+
+    let token: SessionToken = res.json().await;
+    Ok(Uuid::parse_str(&token.token).unwrap())
+}
+
+pub async fn signout_user(
+    _db: &DbPool,
```

Comment:
if not using delete the arg

Reply:
issue with warning as we discussed, i have verified it is used in user tests